src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -721,14 +723,14 @@ export class ResultsViewPageStore {
             this.samples
         ],
         invoke: () => {
-            if (this.genes.result) {
-                return Promise.resolve(_.reduce(_.uniq(this.genes.result.map(g => g.chromosome)), (map: { [chromosome: string]: MobxPromise<CopyNumberSeg[]> }, chromosome: string) => {
+            if (this.referenceGenes.result) {
+                const uniqueReferenceGenes = _.uniq(this.referenceGenes.result.map(g => g.chromosome));
```

Comment:
is it right that we are assigning "chromosome"s to a variable called referenceGenes?

Reply:
we are retrieving copy number segments by chromosomes in reference genome genes.

Reply:
aaron is right, uniqueReferenceGenes contains chromosomes. it should be called uniqueReferenceGeneChromosomes or something like that